[EWL-3604] : adds EC Course Bundle Card

rpama opened the pull request:
<!-- NOTE: Please just put "N/A" for any section below that isn't applicable to the work you've done, do not omit entirely. -->

## Ticket(s)


**Jira Ticket**

- [EWL-3604: EC| Course Bundle Card](https://issues.ama-assn.org/browse/EWL-3604)


## Description

Creates the Course Bundle card


## To Test

- [x] Go to Molecules > EC > EC Bundle Card
- [x] See there's a reasonable implementation of the wireframe


## Relevant Screenshots/GIFs
N/A


## Remaining Tasks

Remaining tasks?


## Additional Notes

Anything more to add?

chelsiejohnston commented:
1. Per the wireframes I'd expect the image to take up the full width of the card. Please either fix or explain why this PR's implementation is better.
![3604-1](https://user-images.githubusercontent.com/12160398/27546912-d18c1be4-5a59-11e7-95c3-9dfdd33c39b5.png)

2. The tablet breakpoint might look more attractive with the text being vertically center-aligned, closer to how it is in the wireframes. Please either fix or explain if you disagree.
![3604-2](https://user-images.githubusercontent.com/12160398/27547079-6bee63fe-5a5a-11e7-9b74-6c184eea462a.png)

3. It looks like the work of including the card row wasn't part of this ticket? Based on the wireframes, the card row will need a different layout than those that exist already. Nothing blocking merge in this point, but I wanted to point it out here. 

LGTM as soon as items 1 and 2 are addressed @rpama :)

chelsiejohnston commented on styleguide/source/_patterns/01-molecules/12-ec/02-ec-bundle-card/_ec-bundle-card.scss:
```diff
@@ -0,0 +1,101 @@
+.ec-bundle-card {
```
The way this class and its files in PatternLab is named doesn't follow our current naming conventions. Since the conventions are loose anyway, and some reorganizing of cards/where things are stored should happen hopefully in the near future, I'm fine with merging as-is for now.

rpama replied:
I'm not sure what you mean here. This is named like the `.hub-card` component.

chelsiejohnston replied:
At the beginning our goal was to name patterns/classes/everything according to BEM when possible. So this would be something like `.card-course-bundle`. I was confused by the class name in this PR because `.ec-bundle-card` wasn't really descriptive enough - at the very least I'd recommend changing this to `.ec-course-bundle-card`. 

Again, I'm okay with merging this in and fixing it later when we reorganize how cards are organized, but just pointing it out for discussion purposes.

chelsiejohnston replied:
Also, can you rename the title in the .md file to "EC Course Bundle Card?"

chelsiejohnston commented on styleguide/source/_patterns/01-molecules/12-ec/02-ec-bundle-card/_ec-bundle-card.scss:
```diff
@@ -0,0 +1,101 @@
+
+  @include breakpoint($bp-med) {
+    // 2-up layout
+    // when there are three items it's 50:50
```
I'm a bit confused at this SCSS being included in this file. This seems to be layout styling for  when there are multiple cards, but there isn't a layout pattern included in this PR. Can you explain by giving some context here? @rpama

rpama replied:
To make it easier for whoever does implement the row? We are not consistent in our card whether the card itself is designed to make a nice row by itself or whether it needs to implement a layout. The current `_card-cta.scss` does a lot of the row layout work, and I used that as the model.

chelsiejohnston replied:
Card patterns should only include styling for the card itself, and then larger patterns that include card patterns should introduce the row CSS. We can talk more about this when we discuss reorganizing the cards, but for now, having this layout/card row CSS in this pattern might actually make my work to implement the cards in the card rows more complicated :| 

If it isn't too much work, can you remove it for now? Again, if you disagree with this, I'm open to discussion when we do that spike.

chelsiejohnston replied:
Actually... on second thought... I think it's fine to leave it this time. I can repurpose it if I need it, or get rid of it if I don't.

chelsiejohnston commented on styleguide/source/_patterns/01-molecules/12-ec/02-ec-bundle-card/ec-bundle-card.md:
```diff
@@ -0,0 +1,11 @@
+---
```
A+, this is a really nice .md :)

rpama commented:
RE: #1  I used padding around the image because that's what the wireframe shows...

![screen shot 2017-06-26 at 11 30 50 am](https://user-images.githubusercontent.com/28711067/27549943-668ffd6a-5a63-11e7-8c7f-ba9506609525.png)

So which one of us is going crazy?

rpama commented:
Have pushed a min height to fix issue #2 about the text alignment. The wireframe isn't really vertically centering. It basically preserves spacing for 2 lines of title whether there are 2 lines of title or not, so I've implemented that spacing now.

chelsiejohnston commented:
1. Looks like in the wires there's padding on mobile, and then the image is full-width at desktop:
![3604-3](https://user-images.githubusercontent.com/12160398/27593544-09c5b47a-5b1d-11e7-9248-ef4c4404e186.png)

2. LGTM.

Please fix 1, as well as consider renaming the .ec-bundle-card namespacing. I'm fine with merging once those are resolved. Thanks!

rpama commented:
- Renamed component to be more in line with Megh's naming of the EC Card Course
 - Added placeholder image since the landscape molecule wasn't cutting it anymore
 - Updated pattern md to reflect new image size requirements
 - Removed padding around the image in the desktop layout
 - Set a max width for the card in the desktop layout

chelsiejohnston commented:
LGTM - thanks for making all the changes!
![bitmoji](https://render.bitstrips.com/v2/cpanel/7897753-201483478_1-s4-v1.png?transparent=1&palette=1&width=246)
